# Estimating Gender Discrimination in the Workplace

In this exercise we’ll use data from the 2018 US Current Population Survey (CPS) to try and estimate the effect of being a woman on workplace compensation. Note that our focus will be only on differential compensation in the work place, and as a result it is important to bear in mind that our estimates are not estimates of all forms of gender discrimination. For example, these analyses will not account for things like gender discrimination in terms of getting jobs.

## Excercise 1

### Explore the data

In [12]:
import pandas as pd

df = pd.read_stata('data/morg18.dta')
df.head()

county  smsastat  age  sex  grade92  race  ethnic  marital  uhourse  \
0       0       1.0   71    1       42     1     NaN        1      NaN   
1       0       1.0   64    2       40     1     NaN        1      NaN   
2       0       1.0   52    2       39     2     NaN        5     40.0   
3       0       1.0   19    2       39     2     NaN        7     40.0   
4       0       1.0   56    2       43     2     NaN        5     40.0   

   earnhre  ...  yrcoll  grprof  gr6cor  ms123  occ2012  \
0      NaN  ...     3.0     NaN     NaN    NaN      NaN   
1      NaN  ...     2.0     NaN     NaN    NaN      NaN   
2   2084.0  ...     NaN     NaN     NaN    NaN   5700.0   
3   1000.0  ...     NaN     NaN     NaN    NaN   5240.0   
4   2500.0  ...     NaN     NaN     NaN    NaN   3255.0   

                        lfsr94               class94  unioncov  \
0  Disabled-Not In Labor Force                                   
1   Retired-Not In Labor Force                                   
2             Employed-At Work    Government - State             
3             Employed-At Work   Private, For Profit        No   
4             Employed-At Work  Government - Federal             

                                               ind02  stfips  
0                                                         AL  
1                                                         AL  
2  Residential care facilities, without nursing (...      AL  
3                   Business support services (5614)      AL  
4                                    Hospitals (622)      AL  

[5 rows x 25 columns]

In [13]:
df.columns

Index(['county', 'smsastat', 'age', 'sex', 'grade92', 'race', 'ethnic',
       'marital', 'uhourse', 'earnhre', 'earnwke', 'chldpres', 'ownchild',
       'ged', 'gedhigr', 'yrcoll', 'grprof', 'gr6cor', 'ms123', 'occ2012',
       'lfsr94', 'class94', 'unioncov', 'ind02', 'stfips'],
      dtype='object')

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 302332 entries, 0 to 302331
Data columns (total 25 columns):
 #   Column    Non-Null Count   Dtype  
---  ------    --------------   -----  
 0   county    302332 non-null  int16  
 1   smsastat  299460 non-null  float64
 2   age       302332 non-null  int8   
 3   sex       302332 non-null  int8   
 4   grade92   302332 non-null  int8   
 5   race      302332 non-null  int8   
 6   ethnic    39222 non-null   float64
 7   marital   302332 non-null  int8   
 8   uhourse   180100 non-null  float64
 9   earnhre   93700 non-null   float64
 10  earnwke   159855 non-null  float64
 11  chldpres  302332 non-null  int8   
 12  ownchild  302332 non-null  int8   
 13  ged       86165 non-null   float64
 14  gedhigr   8193 non-null    float64
 15  yrcoll    82266 non-null   float64
 16  grprof    0 non-null       float64
 17  gr6cor    0 non-null       float64
 18  ms123     0 non-null       float64
 19  occ2012   193103 non-null  float64
 20  lfsr

In [15]:
df.describe()

county       smsastat            age            sex  \
count  302332.000000  299460.000000  302332.000000  302332.000000   
mean       25.626563       1.191885      47.853191       1.521010   
std        61.920623       0.393784      18.749476       0.499559   
min         0.000000       1.000000      16.000000       1.000000   
25%         0.000000       1.000000      32.000000       1.000000   
50%         0.000000       1.000000      48.000000       2.000000   
75%        27.000000       1.000000      63.000000       2.000000   
max       810.000000       2.000000      85.000000       2.000000   

            grade92           race        ethnic        marital  \
count  302332.00000  302332.000000  39222.000000  302332.000000   
mean       40.41597       1.431526      2.569961       3.449734   
std         2.71504       1.275442      2.415717       2.652430   
min        31.00000       1.000000      1.000000       1.000000   
25%        39.00000       1.000000      1.000000       1.000000   
50%        40.00000       1.000000      1.000000       1.000000   
75%        43.00000       1.000000      4.000000       7.000000   
max        46.00000      26.000000      8.000000       7.000000   

             uhourse       earnhre        earnwke       chldpres  \
count  180100.000000  93700.000000  159855.000000  302332.000000   
mean       35.864259   1808.223148     973.528817       1.244827   
std        15.080222   1032.468750     697.560813       2.732001   
min        -4.000000     17.000000       0.000000       0.000000   
25%        35.000000   1150.000000     480.000000       0.000000   
50%        40.000000   1500.000000     775.000000       0.000000   
75%        40.000000   2100.000000    1276.000000       0.000000   
max        99.000000   9999.000000    2884.610000      15.000000   

            ownchild           ged      gedhigr        yrcoll  grprof  gr6cor  \
count  302332.000000  86165.000000  8193.000000  82266.000000     0.0     0.0   
mean        0.459677      1.095085     6.483217      2.809970     NaN     NaN   
std         0.944044      0.293334     1.387283      0.990923     NaN     NaN   
min         0.000000      1.000000     1.000000      1.000000     NaN     NaN   
25%         0.000000      1.000000     6.000000      2.000000     NaN     NaN   
50%         0.000000      1.000000     7.000000      3.000000     NaN     NaN   
75%         0.000000      1.000000     7.000000      3.000000     NaN     NaN   
max        11.000000      2.000000     8.000000      5.000000     NaN     NaN   

       ms123        occ2012  
count    0.0  193103.000000  
mean     NaN    4088.699161  
std      NaN    2601.576064  
min      NaN      10.000000  
25%      NaN    2100.000000  
50%      NaN    4220.000000  
75%      NaN    5620.000000  
max      NaN    9840.000000

## Exercise 2

In [16]:
df = df[df.lfsr94 == 'Employed-At Work']
df = df[df.uhourse >= 35]
df.head()

county  smsastat  age  sex  grade92  race  ethnic  marital  uhourse  \
2        0       1.0   52    2       39     2     NaN        5     40.0   
3        0       1.0   19    2       39     2     NaN        7     40.0   
4        0       1.0   56    2       43     2     NaN        5     40.0   
6       97       1.0   48    1       39     1     NaN        7     40.0   
17      97       1.0   59    1       39     2     NaN        7     40.0   

    earnhre  ...  yrcoll  grprof  gr6cor  ms123  occ2012            lfsr94  \
2    2084.0  ...     NaN     NaN     NaN    NaN   5700.0  Employed-At Work   
3    1000.0  ...     NaN     NaN     NaN    NaN   5240.0  Employed-At Work   
4    2500.0  ...     NaN     NaN     NaN    NaN   3255.0  Employed-At Work   
6    1700.0  ...     NaN     NaN     NaN    NaN   9130.0  Employed-At Work   
17   2000.0  ...     NaN     NaN     NaN    NaN   9620.0  Employed-At Work   

                 class94  unioncov  \
2     Government - State             
3    Private, For Profit        No   
4   Government - Federal             
6    Private, For Profit        No   
17   Private, For Profit        No   

                                                ind02  stfips  
2   Residential care facilities, without nursing (...      AL  
3                    Business support services (5614)      AL  
4                                     Hospitals (622)      AL  
6                          Truck transportation (484)      AL  
17  ****Department stores and discount stores (s45...      AL  

[5 rows x 25 columns]

## Exercise 3

In [17]:
# change from cents to dollars
df['earnhre'] = df['earnhre']/100

In [18]:
df.groupby('sex')['earnhre'].mean()

sex
1    20.554632
2    18.078692
Name: earnhre, dtype: float64

Males earn on average 2 dollars per hour more than women

## Exercise 4

In [19]:
df['annual_wage'] = df['earnhre']*(df['uhourse']*52)

In [20]:
df.groupby('sex')['annual_wage'].mean()

sex
1    45105.302454
2    37864.635158
Name: annual_wage, dtype: float64

In [21]:
print( f"The difference between annual wages between Males and females is ${45105.302454-37864.635158:.4f}" )

The difference between annual wages between Males and females is $7240.6673


## Exercise 5
To have an accurate causal estimate of the effect of being a woman in the work place, both genders must have the same experience and level of seniority. However, in reality, this is not the case since people have worked different years in the field for certain positions and this may affect the estimate. On the other hand, it may also not be true because people negotiate different salaries for work positions.

## Exercise 6

In [24]:
import statsmodels.formula.api as smf

In [25]:
smf.ols('annual_wage ~ C(sex) + age ', df).fit().summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:            annual_wage   R-squared:                       0.057
Model:                            OLS   Adj. R-squared:                  0.057
Method:                 Least Squares   F-statistic:                     1974.
Date:                Tue, 15 Feb 2022   Prob (F-statistic):               0.00
Time:                        14:38:50   Log-Likelihood:            -7.5225e+05
No. Observations:               65755   AIC:                         1.504e+06
Df Residuals:                   65752   BIC:                         1.505e+06
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
===============================================================================
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
Intercept    3.265e+04    288.243    113.284      0.000    3.21e+04    3.32e+04
C(sex)[T.2] -7557.2936    176.107    -42.913      0.000   -7902.462   -7212.125
age           304.1829      6.406     47.485      0.000     291.627     316.738
==============================================================================
Omnibus:                    33771.778   Durbin-Watson:                   1.868
Prob(Omnibus):                  0.000   Jarque-Bera (JB):           307310.621
Skew:                           2.305   Prob(JB):                         0.00
Kurtosis:                      12.535   Cond. No.                         146.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Women on average earn $7557.2936 less than men per year. The raw estimate is less than the calculated regressed difference

## Exercise 7

The comparison we are making is of the annual wages between the two groups. Here we have the indicator groups i.e. females and the reference group is male

## Exercise 8

In [31]:
df["BA"] = df.grade92 >= 43
df["high_school"] = df.grade92 >= 39

In [32]:
df.head()

county  smsastat  age  sex  grade92  race  ethnic  marital  uhourse  \
2        0       1.0   52    2       39     2     NaN        5     40.0   
3        0       1.0   19    2       39     2     NaN        7     40.0   
4        0       1.0   56    2       43     2     NaN        5     40.0   
6       97       1.0   48    1       39     1     NaN        7     40.0   
17      97       1.0   59    1       39     2     NaN        7     40.0   

    earnhre  ...  ms123  occ2012            lfsr94               class94  \
2     20.84  ...    NaN   5700.0  Employed-At Work    Government - State   
3     10.00  ...    NaN   5240.0  Employed-At Work   Private, For Profit   
4     25.00  ...    NaN   3255.0  Employed-At Work  Government - Federal   
6     17.00  ...    NaN   9130.0  Employed-At Work   Private, For Profit   
17    20.00  ...    NaN   9620.0  Employed-At Work   Private, For Profit   

    unioncov                                              ind02  stfips  \
2             Residential care facilities, without nursing (...      AL   
3         No                   Business support services (5614)      AL   
4                                               Hospitals (622)      AL   
6         No                         Truck transportation (484)      AL   
17        No  ****Department stores and discount stores (s45...      AL   

    annual_wage     BA  high_school  
2       43347.2  False         True  
3       20800.0  False         True  
4       52000.0   True         True  
6       35360.0  False         True  
17      41600.0  False         True  

[5 rows x 28 columns]

In [35]:
smf.ols('annual_wage ~ C(sex) + age + C(BA) +C(high_school) ', df).fit().summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:            annual_wage   R-squared:                       0.152
Model:                            OLS   Adj. R-squared:                  0.152
Method:                 Least Squares   F-statistic:                     2946.
Date:                Tue, 15 Feb 2022   Prob (F-statistic):               0.00
Time:                        14:51:54   Log-Likelihood:            -7.4874e+05
No. Observations:               65755   AIC:                         1.497e+06
Df Residuals:                   65750   BIC:                         1.498e+06
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
==========================================================================================
                             coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------
Intercept               2.116e+04    380.335     55.637      0.000    2.04e+04    2.19e+04
C(sex)[T.2]            -9185.1056    168.057    -54.655      0.000   -9514.498   -8855.713
C(BA)[T.True]           1.517e+04    207.215     73.191      0.000    1.48e+04    1.56e+04
C(high_school)[T.True]  9685.9802    293.781     32.970      0.000    9110.169    1.03e+04
age                      309.6835      6.076     50.972      0.000     297.775     321.592
==============================================================================
Omnibus:                    31243.139   Durbin-Watson:                   1.897
Prob(Omnibus):                  0.000   Jarque-Bera (JB):           274123.061
Skew:                           2.098   Prob(JB):                         0.00
Kurtosis:                      12.080   Cond. No.                         234.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

## Exercise 9

The two groups are the indicator groups i.e. annual wages for females with atleast a bachelor's degree or females with atleast a high school degree and the reference group is males with atleast a bachelor's degree or males with atleast a high school degree

## Exercise 10

We see that after controlling for education, the annual wage coefficiant value for females becomes worse. Which implies that education makes women worse off than men